File: Sorting/Comparator.py
from functools import cmp_to_key

class Player:
    def __init__(self, name, score):
        self.name = name
        self.score = score

    def __repr__(self):
        pass 
        
    def comparator(a, b):
        val = b.score - a.score
        if val == 0:
            return -1 if a.name < b.name else 1
        return val
                
n = int(input())
data = []
for i in range(n):
    name, score = input().split()
    score = int(score)
    player = Player(name, score)
    data.append(player)
    
data = sorted(data, key=cmp_to_key(Player.comparator))
for i in data:
    print(i.name, i.score)

File: Sorting/Comparator.java
// Write your Checker class here
class Checker implements Comparator<Player> {
    @Override
    public int compare(Player p1, Player p2) {
        int cmp = Integer.compare(p2.score, p1.score);
        if (cmp != 0) return cmp;
        return p1.name.compareTo(p2.name);
    }
}